typer/core.py

```diff
@@ -360,8 +360,13 @@ def get_help_record(self, ctx: click.Context) -> Optional[Tuple[str, str]]:
         if self.required:
             extra.append(_("required"))
         if extra:
-            extra_str = ";".join(extra)
-            help = f"{help}  [{extra_str}]" if help else f"[{extra_str}]"
+            extra_str = "; ".join(extra)
+            extra_str = f"[{extra_str}]"
+            if rich is not None:
+                # This is needed for when we want to export to HTML
+                extra_str = rich.markup.escape(extra_str).strip()
```

Comment:
This is the only bit that worries me a bit (here and L559) as this changes core functionality anytime `get_help_record` is called on an argument or an option. Existing tests initially failed in `test_doc.py` but I was able to remedy those by using `highlight=False`, `overflow="ignore"` and `crop=False` in the `rich_to_html` helper function.

Nevertheless @tiangolo, it would be good to get your eyes on this 🙏

Reply:
Thank you! Yep, just checked and it's all good! :rocket: